frontend/src/components/body/components/main/components/messageWindow/components/messageHistory/components/dateWiseMessages/components/message/Message.tsx

```diff
@@ -0,0 +1,72 @@
+//Libs
+import React, { useState, useMemo, useRef, useEffect } from "react";
+
+//Components
+import { UserLogo } from '@/icons/userLogo/UserLogo'
+
+//Types
+import { Message as MessageType } from '@/components/body/types/message'
+import { User } from '@/components/body/types/user';
+
+//Styles
+import './message.css'
+
+type Props = {
+    message: MessageType,
+    user?: User,
+    scrollIntoView: boolean,
+}
+
+const getTimeFromTimeStamp = (timestamp: number) => {
+    const hours = new Date(timestamp).getHours();
+    const minutes = ('' + new Date(timestamp).getMinutes());
+    const minutesInProperFormat = (minutes.length === 2 ? minutes : '0' + minutes);
+    return (hours % 12) + ':' + minutesInProperFormat + ' ' + (hours >= 12 ? 'PM' : 'AM');
+}
+
+
+const Message = ({ message, user, scrollIntoView }: Props) => {
+
+    const time = useMemo(() => {
+        return getTimeFromTimeStamp(message.timestamp)
+    }, [message.timestamp]);
+
+    const [hovered, setHovered] = useState(false);
+
+
+    const UserProfilePhoto = (user ? <UserLogo user={user} showStatus={false} /> : null);
```

Comment:
Render directly
Why is this assigned to a constant?

Reply:
okay